components/upload-progress-fullpage.tsx

```diff
@@ -74,11 +77,53 @@ const UploadProgressFullpage: React.FC<Props> = ({ file }) => {
     }
   }, [upload]);
 
+
+  const startFileValidation = (_file: File) => {
+    // Attempt to load the file as a video element and inspect its duration
+    // metadata. This is not an authoritative check of video duration, but
+    // rather intended to serve as just a simple and fast sanity check.
+    if (!_file.type.includes("video")) {
+      console.warn(`file type (${_file.type}) does not look like video!`);
+      setInputValidated(true);
+      return;
+    }
+
+    const video = document.createElement('video');
+    video.preload = 'metadata';
+    video.onloadedmetadata = function() {
+      URL.revokeObjectURL(video.src);
+      if (video.duration > MAX_VIDEO_DURATION_SEC) {
+        setValidationError(`file duration (${video.duration.toString()}s) exceeds allowed maximum (${MAX_VIDEO_DURATION_SEC}s)!`);
+        return;
+      }
+      setInputValidated(true);
+    };
+    video.onerror = function() {
+      // The file has a video MIME type, but we were unable to load its
+      // metadata for some reason.
+      console.warn("failed to load video file metadata for validation!");
+      URL.revokeObjectURL(video.src);
+      setInputValidated(true);
+    };
+    video.src = URL.createObjectURL(file);
```

Comment:
Is there ever a case where neither `onloadedmetadata` nor `onerror` fire? That would leave us in a broken state that never resolves.

I'm thinking of cases where the browser blocks autoplay or something like that. Let's make sure about that.

I'm also wondering about cases when setting `video.src` with a src file that the browser doesn't know how to play natively, for example a codec that is not supported. I suspect `video.onerror` would fire, but we should make sure.

Reply:
It seems like metdata will either load or an error will be thrown, but the docs I've found haven't really discussed the video element lifecycle in that way (they kind of just list/describe all the possible event types). Would it be reasonable to just add a timeout of some sort? Or a better resource you'd recommend? I've been basically looking at https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Media_events

I've tested this with non-video files and they always fire `onerror`. Not sure what you mean re: autoplay, would that prevent the video element from getting created?

Reply:
My concerns were around behavior I've seen like this:

https://github.com/muxinc/stream.new/blob/main/components/video-player.tsx#L14-L19

For example mobile safari was not firing any events until the user interacted with the element